common/buildcraft/lib/misc/StackUtil.java

```diff
@@ -298,6 +304,12 @@ public static boolean isMatchingItem(@Nonnull final ItemStack base, @Nonnull fin
         if (base.getItem() != comparison.getItem()) {
             return false;
         }
+
+        Set<String> matchingNBT = matchingNBTRegistry.get(base.getItem());
+        if (matchingNBT != null && !isNbtSubsetEquals(base, comparison, matchingNBT)) {
+            return false;
+        }
+
```

Comment:
If I will change the NBT set to a predicate, then this condition and the `matchNBT` are not exactly the opposite. The predicate can match for anything in `ItemStack`, for example, a name of an item. I think it would be better to place my code not in the else branch but after the `matchingNBT` block.

Reply:
As I understand, if NBTs are equals, then there is no other way to tell the difference between item stacks. So, yes, it must be in else block to prevent redundant checks. But maybe I'm wrong, I am not good at Minecraft API.

Reply:
Yes, ItemStack's only have three fields that we care about: Item, metadata/damage (int), and NBT.

(Items also can have capabilities but as BC doesn't handle them at all it's probably not a good idea to start caring about them in this PR).

Reply:
Thanks!